adapters/beachfront/beachfront.go

```diff
@@ -197,289 +188,433 @@ func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *a
+		if request.Device.ConnectionType != nil {
+			ctype := *request.Device.ConnectionType
+			device.ConnectionType = &ctype
+		}
 
-	for i := 0; i < len(request.Imp); i++ {
+		if request.Device.Geo != nil {
+			deviceGeo := *request.Device.Geo
+			device.Geo = &deviceGeo
+		}
 
-		beachfrontExt, err := getBeachfrontExtension(request.Imp[i])
+		if request.Device.DNT != nil {
+			dnt := *request.Device.DNT
+			device.DNT = &dnt
+		}
 
-		if err != nil {
-			errs = append(errs, err)
-			continue
+		if request.Device.Lmt != nil {
+			lmt := *request.Device.Lmt
+			device.Lmt = &lmt
 		}
 
-		appid, err := getAppId(beachfrontExt, openrtb_ext.BidTypeBanner)
+		requestCopy.Device = &device
+	}
 
-		if err != nil {
-			// Failed to get an appid, so this request is junk.
-			errs = append(errs, err)
-			continue
-		}
+	if request.User != nil {
+		user := *request.User
 
-		slot := beachfrontSlot{
-			Id:       appid,
-			Slot:     request.Imp[i].ID,
-			Bidfloor: beachfrontExt.BidFloor,
+		if request.User.Geo != nil {
+			userGeo := *request.User.Geo
+			user.Geo = &userGeo
 		}
 
-		if beachfrontExt.BidFloor <= minBidFloor {
-			slot.Bidfloor = 0
-		}
+		requestCopy.User = &user
+	}
 
-		for j := 0; j < len(request.Imp[i].Banner.Format); j++ {
+	if request.Source != nil {
+		source := *request.Source
+		requestCopy.Source = &source
+	}
 
-			slot.Sizes = append(slot.Sizes, beachfrontSize{
-				H: request.Imp[i].Banner.Format[j].H,
-				W: request.Imp[i].Banner.Format[j].W,
-			})
-		}
+	if request.Regs != nil {
+		regs := *request.Regs
+		requestCopy.Regs = &regs
+	}
 
-		bfr.Slots = append(bfr.Slots, slot)
+	return requestCopy
+}
+
+func getBannerRequest(request *openrtb.BidRequest, errs []error) (bannerRequest, []error) {
+	var secure int8
+	var bannerReq bannerRequest
+	bannerReq, secure, errs = impsToSlots(request.Imp, errs)
+
+	if len(bannerReq.Slots) == 0 {
+		errs = append(errs, errors.New("unable to construct a valid banner request"))
+		return bannerReq, errs
 	}
 
-	if len(bfr.Slots) == 0 {
-		return bfr, errs
+	if request.Site != nil && request.Site.Page != "" {
+		bannerReq.Secure = isPageSecure(request.Site.Page)
+	} else {
+		bannerReq.Secure = secure
 	}
 
 	if request.Device != nil {
-		bfr.IP = getIP(request.Device.IP)
-		bfr.DeviceModel = request.Device.Model
-		bfr.DeviceOs = request.Device.OS
+		bannerReq.IP = getIP(request.Device.IP)
+		bannerReq.DeviceModel = request.Device.Model
+		bannerReq.DeviceOs = request.Device.OS
 		if request.Device.DNT != nil {
-			bfr.Dnt = *request.Device.DNT
+			bannerReq.Dnt = *request.Device.DNT
 		}
 		if request.Device.UA != "" {
-			bfr.UA = request.Device.UA
+			bannerReq.UA = request.Device.UA
 		}
 	}
 
 	var t = fallBackDeviceType(request)
 
 	if t == openrtb.DeviceTypeMobileTablet {
-		bfr.Page = request.App.Bundle
+		bannerReq.Page = request.App.Bundle
 		if request.App.Domain == "" {
-			bfr.Domain = getDomain(request.App.Domain)
+			bannerReq.Domain = getDomain(request.App.Domain)
```

Comment:
Shouldn't this be `getDomain(request.App.Bundle)` or something. As current, you are calling `getDomain("")` and should get back `""`, making this a noop.

Reply:
Right. Changed to this locally:

`
		if request.App.Domain != "" {
			bannerReq.Domain = request.App.Domain
		}`